Segmentaion fault when using -u

```
$ kube-capacity -pu
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x10cd9f2]

goroutine 1 [running]:
github.com/robscott/kube-capacity/pkg/capacity.(*clusterMetric).addPodMetric(0xc000290a38, 0xc0002914f0, 0x0, 0x0, 0x0, 0x0, 0xc0006bda80, 0x1b, 0x0, 0x0, ...)
        /Users/rob/go/src/github.com/robscott/kube-capacity/pkg/capacity/resources.go:175 +0x932
github.com/robscott/kube-capacity/pkg/capacity.buildClusterMetric(0xc000136930, 0xc000f748c0, 0xc000139d50, 0x0, 0x0, 0x0)
        /Users/rob/go/src/github.com/robscott/kube-capacity/pkg/capacity/resources.go:105 +0x82e
github.com/robscott/kube-capacity/pkg/capacity.FetchAndPrint(0x1010100, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x13670c5, ...)
        /Users/rob/go/src/github.com/robscott/kube-capacity/pkg/capacity/capacity.go:50 +0x216
github.com/robscott/kube-capacity/pkg/cmd.glob..func1(0x20c18a0, 0xc000370840, 0x0, 0x1)
        /Users/rob/go/src/github.com/robscott/kube-capacity/pkg/cmd/root.go:49 +0x21d
github.com/robscott/kube-capacity/vendor/github.com/spf13/cobra.(*Command).execute(0x20c18a0, 0xc0000b2030, 0x1, 0x1, 0x20c18a0, 0xc0000b2030)
        /Users/rob/go/src/github.com/robscott/kube-capacity/vendor/github.com/spf13/cobra/command.go:766 +0x2ae
github.com/robscott/kube-capacity/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x20c18a0, 0xc000427f68, 0x10e368e, 0x20c18a0)
        /Users/rob/go/src/github.com/robscott/kube-capacity/vendor/github.com/spf13/cobra/command.go:852 +0x2ec
github.com/robscott/kube-capacity/vendor/github.com/spf13/cobra.(*Command).Execute(...)
        /Users/rob/go/src/github.com/robscott/kube-capacity/vendor/github.com/spf13/cobra/command.go:800
github.com/robscott/kube-capacity/pkg/cmd.Execute()
        /Users/rob/go/src/github.com/robscott/kube-capacity/pkg/cmd/root.go:79 +0x32
main.main()
        /Users/rob/go/src/github.com/robscott/kube-capacity/main.go:22 +0x20

```
```
$ kube-capacity -u
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x10cd9f2]

goroutine 1 [running]:
github.com/robscott/kube-capacity/pkg/capacity.(*clusterMetric).addPodMetric(0xc0004f0a38, 0xc0004f14f0, 0x0, 0x0, 0x0, 0x0, 0xc00061b300, 0x1b, 0x0, 0x0, ...)
        /Users/rob/go/src/github.com/robscott/kube-capacity/pkg/capacity/resources.go:175 +0x932
github.com/robscott/kube-capacity/pkg/capacity.buildClusterMetric(0xc000102930, 0xc000933c70, 0xc00011ecb0, 0x0, 0x0, 0x0)
        /Users/rob/go/src/github.com/robscott/kube-capacity/pkg/capacity/resources.go:105 +0x82e
github.com/robscott/kube-capacity/pkg/capacity.FetchAndPrint(0x1010000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x13670c5, ...)
        /Users/rob/go/src/github.com/robscott/kube-capacity/pkg/capacity/capacity.go:50 +0x216
github.com/robscott/kube-capacity/pkg/cmd.glob..func1(0x20c18a0, 0xc000344860, 0x0, 0x1)
        /Users/rob/go/src/github.com/robscott/kube-capacity/pkg/cmd/root.go:49 +0x21d
github.com/robscott/kube-capacity/vendor/github.com/spf13/cobra.(*Command).execute(0x20c18a0, 0xc00000c0b0, 0x1, 0x1, 0x20c18a0, 0xc00000c0b0)
        /Users/rob/go/src/github.com/robscott/kube-capacity/vendor/github.com/spf13/cobra/command.go:766 +0x2ae
github.com/robscott/kube-capacity/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x20c18a0, 0xc0003fbf68, 0x10e368e, 0x20c18a0)
        /Users/rob/go/src/github.com/robscott/kube-capacity/vendor/github.com/spf13/cobra/command.go:852 +0x2ec
github.com/robscott/kube-capacity/vendor/github.com/spf13/cobra.(*Command).Execute(...)
        /Users/rob/go/src/github.com/robscott/kube-capacity/vendor/github.com/spf13/cobra/command.go:800
github.com/robscott/kube-capacity/pkg/cmd.Execute()
        /Users/rob/go/src/github.com/robscott/kube-capacity/pkg/cmd/root.go:79 +0x32
main.main()
        /Users/rob/go/src/github.com/robscott/kube-capacity/main.go:22 +0x20
```
```
$ kube-capacity version
kube-capacity version 0.3.1
```

Metrics-server is working for me with kubectl:
```
$ kubectl top node 
NAME    CPU(cores)   CPU%   MEMORY(bytes)   MEMORY%   
h1355   1693m        21%    5304Mi          67%       
h1879   2835m        35%    5200Mi          66%       
h230    3412m        42%    4765Mi          60%       
h303    1792m        22%    5162Mi          65%       
h504    2599m        32%    4739Mi          60%       
h5345   79m          0%     11161Mi         46%       
h71     1598m        19%    11731Mi         73%       
h783    1161m        14%    4008Mi          50%       
h834    1263m        15%    5533Mi          70%       
h911    1763m        22%    5406Mi          68%       
s234    839m         10%    3980Mi          50%       
s237    975m         12%    5451Mi          69%       
s238    399m         4%     2985Mi          37%       
s239    526m         6%     3192Mi          40%       
```
` GET https://api.example.com:8443/apis/metrics.k8s.io/v1beta1/nodes  200 OK in 98 milliseconds`

```
$ kubectl version --short
Client Version: v1.15.0-alpha.3
Server Version: v1.14.1
```

--- a/pkg/capacity/capacity.go
+++ b/pkg/capacity/capacity.go
@@ -37,6 +37,8 @@ func FetchAndPrint(showContainers, showPods, showUtil bool, podLabels, nodeLabel
 
 	podList, nodeList := getPodsAndNodes(clientset, podLabels, nodeLabels, namespaceLabels)
 	pmList := &v1beta1.PodMetricsList{}
+	nmList := &v1beta1.NodeMetricsList{}
+
 	if showUtil {
 		mClientset, err := kube.NewMetricsClientSet(kubeContext)
 		if err != nil {
@@ -44,10 +46,11 @@ func FetchAndPrint(showContainers, showPods, showUtil bool, podLabels, nodeLabel
 			os.Exit(4)
 		}
 
-		pmList = getMetrics(mClientset)
+		pmList = getPodMetrics(mClientset)
+		nmList = getNodeMetrics(mClientset)
 	}
 
-	cm := buildClusterMetric(podList, pmList, nodeList)
+	cm := buildClusterMetric(podList, pmList, nodeList, nmList)
 	printList(&cm, showContainers, showPods, showUtil, output, sortBy)
 }
 
@@ -115,7 +118,7 @@ func getPodsAndNodes(clientset kubernetes.Interface, podLabels, nodeLabels, name
 	return podList, nodeList
 }
 
-func getMetrics(mClientset *metrics.Clientset) *v1beta1.PodMetricsList {
+func getPodMetrics(mClientset *metrics.Clientset) *v1beta1.PodMetricsList {
 	pmList, err := mClientset.MetricsV1beta1().PodMetricses("").List(metav1.ListOptions{})
 	if err != nil {
 		fmt.Printf("Error getting Pod Metrics: %v\n", err)
@@ -124,4 +127,15 @@ func getMetrics(mClientset *metrics.Clientset) *v1beta1.PodMetricsList {
 	}
 
 	return pmList
+}
+
+func getNodeMetrics(mClientset *metrics.Clientset) *v1beta1.NodeMetricsList {
+	nmList, err := mClientset.MetricsV1beta1().NodeMetricses().List(metav1.ListOptions{})
+	if err != nil {
+		fmt.Printf("Error getting Node Metrics: %v\n", err)
+		fmt.Println("For this to work, metrics-server needs to be running in your cluster")
+		os.Exit(7)
+	}
+
+	return nmList
 }

--- a/pkg/capacity/list_test.go
+++ b/pkg/capacity/list_test.go
@@ -261,6 +261,18 @@ func getTestClusterMetric() clusterMetric {
 					},
 				},
 			},
+		}, &v1beta1.NodeMetricsList{
+			Items: []v1beta1.NodeMetrics{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "example-node-1",
+					},
+					Usage: corev1.ResourceList{
+						"cpu":    resource.MustParse("63m"),
+						"memory": resource.MustParse("439Mi"),
+					},
+				},
+			},
 		},
 	)
 }

--- a/pkg/capacity/resources.go
+++ b/pkg/capacity/resources.go
@@ -70,7 +70,8 @@ type containerMetric struct {
 	memory *resourceMetric
 }
 
-func buildClusterMetric(podList *corev1.PodList, pmList *v1beta1.PodMetricsList, nodeList *corev1.NodeList) clusterMetric {
+func buildClusterMetric(podList *corev1.PodList, pmList *v1beta1.PodMetricsList,
+	nodeList *corev1.NodeList, nmList *v1beta1.NodeMetricsList) clusterMetric {
 	cm := clusterMetric{
 		cpu:         &resourceMetric{resourceType: "cpu"},
 		memory:      &resourceMetric{resourceType: "memory"},
@@ -90,9 +91,11 @@ func buildClusterMetric(podList *corev1.PodList, pmList *v1beta1.PodMetricsList,
 			},
 			podMetrics: map[string]*podMetric{},
 		}
+	}
 
-		cm.cpu.allocatable.Add(node.Status.Allocatable["cpu"])
-		cm.memory.allocatable.Add(node.Status.Allocatable["memory"])
+	for _, nm := range nmList.Items {
+		cm.nodeMetrics[nm.Name].cpu.utilization = nm.Usage["cpu"]
+		cm.nodeMetrics[nm.Name].memory.utilization = nm.Usage["memory"]
 	}
 
 	podMetrics := map[string]v1beta1.PodMetrics{}
@@ -103,6 +106,13 @@ func buildClusterMetric(podList *corev1.PodList, pmList *v1beta1.PodMetricsList,
 	for _, pod := range podList.Items {
 		if pod.Status.Phase != corev1.PodSucceeded && pod.Status.Phase != corev1.PodFailed {
 			cm.addPodMetric(&pod, podMetrics[fmt.Sprintf("%s-%s", pod.GetNamespace(), pod.GetName())])
+		}
+	}
+
+	for _, node := range nodeList.Items {
+		nm := cm.nodeMetrics[node.Name]
+		if nm != nil {
+			cm.addNodeMetric(cm.nodeMetrics[node.Name])
 		}
 	}
 
@@ -156,11 +166,6 @@ func (cm *clusterMetric) addPodMetric(pod *corev1.Pod, podMetrics v1beta1.PodMet
 	}
 
 	if nm != nil {
-		cm.cpu.request.Add(req["cpu"])
-		cm.cpu.limit.Add(limit["cpu"])
-		cm.memory.request.Add(req["memory"])
-		cm.memory.limit.Add(limit["memory"])
-
 		nm.podMetrics[key] = pm
 		nm.podMetrics[key].cpu.allocatable = nm.cpu.allocatable
 		nm.podMetrics[key].memory.allocatable = nm.memory.allocatable
@@ -172,20 +177,13 @@ func (cm *clusterMetric) addPodMetric(pod *corev1.Pod, podMetrics v1beta1.PodMet
 	}
 
 	for _, container := range podMetrics.Containers {
-		pm.containerMetrics[container.Name].cpu.utilization = container.Usage["cpu"]
-		pm.cpu.utilization.Add(container.Usage["cpu"])
-		pm.containerMetrics[container.Name].memory.utilization = container.Usage["memory"]
-		pm.memory.utilization.Add(container.Usage["memory"])
-
-		if nm == nil {
-			continue
+		cm := pm.containerMetrics[container.Name]
+		if cm != nil {
+			pm.containerMetrics[container.Name].cpu.utilization = container.Usage["cpu"]
+			pm.cpu.utilization.Add(container.Usage["cpu"])
+			pm.containerMetrics[container.Name].memory.utilization = container.Usage["memory"]
+			pm.memory.utilization.Add(container.Usage["memory"])
 		}
-
-		nm.cpu.utilization.Add(container.Usage["cpu"])
-		nm.memory.utilization.Add(container.Usage["memory"])
-
-		cm.cpu.utilization.Add(container.Usage["cpu"])
-		cm.memory.utilization.Add(container.Usage["memory"])
 	}
 }
 

--- a/pkg/capacity/resources_test.go
+++ b/pkg/capacity/resources_test.go
@@ -28,7 +28,7 @@ import (
 
 func TestBuildClusterMetricEmpty(t *testing.T) {
 	cm := buildClusterMetric(
-		&corev1.PodList{}, &v1beta1.PodMetricsList{}, &corev1.NodeList{},
+		&corev1.PodList{}, &v1beta1.PodMetricsList{}, &corev1.NodeList{}, &v1beta1.NodeMetricsList{},
 	)
 
 	expected := clusterMetric{
@@ -129,6 +129,18 @@ func TestBuildClusterMetricFull(t *testing.T) {
 					},
 				},
 			},
+		}, &v1beta1.NodeMetricsList{
+			Items: []v1beta1.NodeMetrics{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "example-node-1",
+					},
+					Usage: corev1.ResourceList{
+						"cpu":    resource.MustParse("43m"),
+						"memory": resource.MustParse("349Mi"),
+					},
+				},
+			},
 		},
 	)
 
@@ -136,14 +148,14 @@ func TestBuildClusterMetricFull(t *testing.T) {
 		allocatable: resource.MustParse("1000m"),
 		request:     resource.MustParse("350m"),
 		limit:       resource.MustParse("400m"),
-		utilization: resource.MustParse("23m"),
+		utilization: resource.MustParse("43m"),
 	}
 
 	memoryExpected := &resourceMetric{
 		allocatable: resource.MustParse("4000Mi"),
 		request:     resource.MustParse("400Mi"),
 		limit:       resource.MustParse("700Mi"),
-		utilization: resource.MustParse("299Mi"),
+		utilization: resource.MustParse("349Mi"),
 	}
 
 	assert.NotNil(t, cm.cpu)

--- a/pkg/capacity/table.go
+++ b/pkg/capacity/table.go
@@ -64,10 +64,13 @@ func (tp *tablePrinter) Print() {
 
 	if len(sortedNodeMetrics) > 1 {
 		tp.printClusterLine()
-		tp.printLine(&tableLine{})
 	}
 
 	for _, nm := range sortedNodeMetrics {
+		if tp.showPods || tp.showContainers {
+			tp.printLine(&tableLine{})
+		}
+
 		tp.printNodeLine(nm.name, nm)
 
 		if tp.showPods || tp.showContainers {

--- a/pkg/cmd/version.go
+++ b/pkg/cmd/version.go
@@ -28,6 +28,6 @@ var versionCmd = &cobra.Command{
 	Use:   "version",
 	Short: "Print the version number of kube-capacity",
 	Run: func(cmd *cobra.Command, args []string) {
-		fmt.Println("kube-capacity version 0.3.1")
+		fmt.Println("kube-capacity version 0.3.2")
 	},
 }

